### ISSUE TITLE
streams3

### ISSUE THREAD

**Issue by substack** [substack/node-browserify#1312]
To upgrade to streams3 (readable-stream ^2) we'll need to use the next major version of stream-browserify, which has already been published. Unfortunately this upgrade causes some strange cyclic dependency behavior in the browserify crypto tests (test/crypto.js and test/crypto_ig.js): https://github.com/substack/stream-browserify/pull/9#issuecomment-115370040

```
TypeError: evalmachine.<anonymous>:15118
          ctor.prototype = Object.create(superCtor.prototype, {
                                                  ^
      Cannot read property 'prototype' of undefined
```

Strangely, I don't get this error invoking browserify directly:

```
$ browserify <(echo "console.log(require('crypto').createHash('sha256').update('whatever').digest('hex'))") | node
85738f8f9a7f1b04b5329c590ebcb9e425925c6d0984089c43a022de4f19c281
$ browserify <(echo "console.log(require('crypto').createHash('sha256').update('whatever').digest('hex'))") | browser-run
85738f8f9a7f1b04b5329c590ebcb9e425925c6d0984089c43a022de4f19c281
^C
```

**Comment by feross** [comment]
The crypto tests pass if I remove `require: function () {}` from [this line](https://github.com/substack/node-browserify/blob/f9c256174fe282f7a8ca619d7168161e8e208524/test/crypto.js#L35).

What is the purpose of that line?

**Comment by feross** [comment]
The tests pass now thanks to this: https://github.com/nodejs/readable-stream/pull/153

**Comment by feross** [comment]
This is released in browserify 11.0.0.
